data-sets-api-server/src/main/java/org/zowe/unix/files/services/zosmf/DeleteUnixFileRunner.java

```diff
@@ -75,12 +75,12 @@ protected ZoweApiRestException createException(JsonObject jsonResponse, int stat
                 }
             }
             if (statusCode == HttpStatus.SC_INTERNAL_SERVER_ERROR) {
-            	if (details.toString().contains("EDC5111I Permission denied.")) {
+                if (details.toString().contains("EDC5111I Permission denied.")) {
                     throw new PermissionDeniedFileException(path);
                 } else if (details.toString().contains("EDC5136I Directory not empty.")) {
```

Comment:
The directory not empty exception is the only special exception case here so should be the only one handled by the  createException function in this class, we should call the super createException function from AbstractZosmfUnixFilesRequestRunner to handle the generic cases such as permission denied, not found, etc.

Reply:
PermissionDeniedFileException is kind of specific - no write permission
otherone i removed and now using createUnixFileException as suggested

Reply:
removed permission denied using createUnixFileException for all other cases except empty directory